pkg/forman/forman.go

```diff
@@ -0,0 +1,101 @@
+package forman
+
+import (
+	"time"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/utils"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/workers"
+
+	"go.uber.org/zap"
+)
+
+type provider struct {
+	working    bool
+	next       *time.Time
+	alarmClock *time.Timer
+}
+
+func GoForman(
+	logger *zap.Logger,
+	client *db.Database,
+	workerCount int,
+	reconcile func(*zap.Logger, *db.Database, types.Request) types.Result,
+) chan types.Request {
+	requests := make(chan types.Request)
+
+	go func() {
+		toWorkers := make(chan types.Request)
+		results := make(chan utils.Pair[types.Request, types.Result])
+		providers := map[string]*provider{}
+
+		sendToWorkers := func(req types.Request) {
+			go func() { toWorkers <- req }()
+		}
+
+		logger.Info("Starting", zap.Int("workers", workerCount))
+		for i := 0; i < workerCount; i++ {
+			workers.GoWorker(toWorkers, results, func(req types.Request) types.Result {
+				return reconcile(logger, client, req)
+			})
```

Comment:
Please see prev review for reasoning 
```suggestion
			workers.GoWorker(toWorkers, results, reconcile)
```

Reply:
This suggest is not working, it needs more changes in the worker cod and the question i have is who will send the dbclient and logger instance here?

Reply:
I am closing this thread as I mentioned how I would like this to look like (with code suggestions) on a newer review